File: src/pratice/pointtopoint/두용액2470.java
package pratice.pointtopoint;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class 두용액2470 {
    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static int N;
    public static int[] A;

    public static void main(String[] args) throws IOException {
        input();
        Arrays.sort(A);
        pro();
    }

    private static void input() throws IOException {
        N = Integer.parseInt(br.readLine());
        A = new int[N];
        String[] split = br.readLine().split(" ");

        for (int i = 0; i < N; i++) {
            A[i] = Integer.parseInt(split[i]);
        }
    }

    private static void pro() {
       int L = 0;
       int R = N - 1;
       int v1 = 0;
       int v2 = 0;
       int bestSum = Integer.MAX_VALUE;
       while(L < R){
           int sum = A[L] + A[R];
           if(Math.abs(sum) < bestSum){
               bestSum = Math.abs(sum);
               v1 = A[L];
               v2 = A[R];
           }
            if(sum > 0){
                R--;
            }else L++;
       }
        System.out.println(v1 + " " + v2);
    }

}

File: src/pratice/pointtopoint/두용액2470.kt
package pratice.pointtopoint

import kotlin.math.abs

fun main() {
    val N = readln().toInt()
    val arr = readln().split(" ").map { it.toInt() }.toMutableList()
    arr.sort()

    var L = 0
    var R = N - 1
    var v1 = 0
    var v2 = 0
    var bestNum = Int.MAX_VALUE
    while (L < R) {
        val sum = arr[L] + arr[R]
        if (abs(sum) < bestNum) {
            bestNum = abs(sum)
            v1 = arr[L]
            v2 = arr[R]
        }
        if (sum > 0) {
            R--
        } else {
            L++
        }
    }
    println("$v1  $v2")
}
